packages/core/src/modules/waas/classes/waas-service-tree-node.class.ts

```diff
@@ -0,0 +1,93 @@
+import { IRequestConfigInternal, IWaasHooks, toArray } from '../interfaces';
+import { WaasEventEmitter } from './waas-event-emitter.class';
+
+export interface PathElement {
+  path: string;
+  variables: string[];
+}
+
+export class WaasServiceTreeNode extends WaasEventEmitter {
+  path: string;
+  variables: string[] = [];
+  parentNode: WaasServiceTreeNode = null;
+  children: { [key: string]: WaasServiceTreeNode } = {};
+  hooks: IWaasHooks = { after: [], before: [] };
+
+  constructor(parent: WaasServiceTreeNode = null) {
+    super(parent);
+    this.parentNode = parent;
+  }
+
+  static isVariable(path: string): boolean {
+    return (path || '').length === 0 ? false : path[0] === ':';
+  }
```

Comment:
Is to complicated
static isVariable(path: string = ''): boolean {
         return path.charAt(0) === ':';
}
or path.indexOf(':') === 0

Reply:
done